records_mover/db/postgres/postgres_copy_options.py

```diff
@@ -0,0 +1,374 @@
+    #  is \N (backslash-N) in text format, and an unquoted empty
+    #  string in CSV format. You might prefer an empty string even in
+    #  text format for cases where you don't want to distinguish nulls
+    #  from empty strings. This option is not allowed when using
+    #  binary format.
+    #
+    #  Note: When using COPY FROM, any data item that matches this
+    #  string will be stored as a null value, so you should make sure
+    #  that you use the same string as you used with COPY TO.
+    #
+    # HEADER
+    #
+    #  Specifies that the file contains a header line with the names
+    #  of each column in the file. On output, the first line contains
+    #  the column names from the table, and on input, the first line
+    #  is ignored. This option is allowed only when using CSV format.
+    #
+    if hints['header-row']:
+        cant_handle_hint(fail_if_cant_handle_hint, 'header-row', hints)
+    else:
+        quiet_remove(unhandled_hints, 'header-row')
+
+    # QUOTE
+    #
+    #  Specifies the quoting character to be used when a data value is
+    #  quoted. The default is double-quote. This must be a single
+    #  one-byte character. This option is allowed only when using CSV
+    #  format.
+    #
+
+    quiet_remove(unhandled_hints, 'quotechar')
+
+    # ESCAPE
+    #
+    #  Specifies the character that should appear before a data
+    #  character that matches the QUOTE value. The default is the same
+    #  as the QUOTE value (so that the quoting character is doubled if
+    #  it appears in the data). This must be a single one-byte
+    #  character. This option is allowed only when using CSV format.
+    #
+
+    if hints['doublequote']:
+        cant_handle_hint(fail_if_cant_handle_hint, 'doublequote', hints)
+    else:
+        quiet_remove(unhandled_hints, 'doublequote')
+
+    # FORCE_QUOTE
+    #
+    #  Forces quoting to be used for all non-NULL values in each
+    #  specified column. NULL output is never quoted. If * is
+    #  specified, non-NULL values will be quoted in all columns. This
+    #  option is allowed only in COPY TO, and only when using CSV
+    #  format.
+    #
+
+    if hints['quoting'] is not None:
+        cant_handle_hint(fail_if_cant_handle_hint, 'quoting', hints)
+    else:
+        quiet_remove(unhandled_hints, 'quoting')
+
+    # FORCE_NOT_NULL
+    #
+    #  Do not match the specified columns' values against the null
+    #  string. In the default case where the null string is empty,
+    #  this means that empty values will be read as zero-length
+    #  strings rather than nulls, even when they are not quoted. This
+    #  option is allowed only in COPY FROM, and only when using CSV
+    #  format.
+    #
+    # ENCODING
+    #
+    #  Specifies that the file is encoded in the encoding_name. If
+    #  this option is omitted, the current client encoding is
+    #  used. See the Notes below for more details.
+
+    # TODO: Dedupe common params
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker